cmd/kueue/main.go

```diff
@@ -163,6 +168,24 @@ func main() {
 	// Controllers who register after manager starts will start directly.
 	go setupControllers(mgr, cCache, queues, certsReady, &cfg, serverVersionFetcher)
 
+	if features.Enabled(features.ProvisioningACC) {
+		// setup provision controller
+		go func() {
+			if !provisioning.ServerSupportsProvisioningRequest(mgr) {
+				setupLog.Info("Provisioning Requests are not supported, skip admission check controller")
```

Comment:
Should be a warning.

Reply:
We don't use Warning in k8s https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/migration-to-structured-logging.md#change-log-functions

Reply:
We do, if it is not a structure log. K/K has still plenty of klog.warning. Anyway, If provisioning request is not supported and provisioningcontroller is on, we should have something more than info. So let it be a fatal error.

Reply:
> We do, if it is not a structure log

That's legacy code that we are cleaning up.

I much rather do an error and exit

If we are going to error, we should have a permanent configuration (not just feature gate) that allows to disable the feature, and probably disabled by default, as not every cluster has a cluster autoscaler.

Reply:
Eventually, yes, we need a flag/field like with integrations. I guess for now/this PR it may be just a featuregate.